hive/src/main/java/com/netflix/iceberg/hive/HiveTableOperations.java

```diff
@@ -0,0 +1,224 @@
+ * limitations under the License.
+ */
+package com.netflix.iceberg.hive;
+
+import com.google.common.collect.Lists;
+import com.netflix.iceberg.BaseMetastoreTableOperations;
+import com.netflix.iceberg.Schema;
+import com.netflix.iceberg.TableMetadata;
+import com.netflix.iceberg.exceptions.CommitFailedException;
+import com.netflix.iceberg.exceptions.NoSuchTableException;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.LockComponent;
+import org.apache.hadoop.hive.metastore.api.LockLevel;
+import org.apache.hadoop.hive.metastore.api.LockRequest;
+import org.apache.hadoop.hive.metastore.api.LockResponse;
+import org.apache.hadoop.hive.metastore.api.LockState;
+import org.apache.hadoop.hive.metastore.api.LockType;
+import org.apache.hadoop.hive.metastore.api.NoSuchObjectException;
+import org.apache.hadoop.hive.metastore.api.SerDeInfo;
+import org.apache.hadoop.hive.metastore.api.SerdeType;
+import org.apache.hadoop.hive.metastore.api.StorageDescriptor;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.thrift.TException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+/**
+ * TODO we should be able to extract some more commonalities to BaseMetastoreTableOperations to
+ * avoid code duplication between this class and Metacat Tables.
+ */
+public class HiveTableOperations extends BaseMetastoreTableOperations {
+  private static final Logger LOG = LoggerFactory.getLogger(HiveTableOperations.class);
+
+  private final IMetaStoreClient metaStoreClient;
+  private final String database;
+  private final String tableName;
+
+  protected HiveTableOperations(Configuration conf, IMetaStoreClient metaStoreClient, String database, String table) {
+    super(conf);
+    this.metaStoreClient = metaStoreClient;
+    this.database = database;
+    tableName = table;
+    refresh();
+  }
+
+  @Override
+  public TableMetadata refresh() {
+    try {
+      final Table table = metaStoreClient.getTable(database, tableName);
+      String tableType = table.getParameters().get(TABLE_TYPE_PROP);
+
+      if (tableType == null || !tableType.equalsIgnoreCase(ICEBERG_TABLE_TYPE_VALUE)) {
+        throw new IllegalArgumentException(format("Invalid tableName, not Iceberg: %s.%s", database, table));
+      }
+
+      String metadataLocation = table.getParameters().get(METADATA_LOCATION_PROP);
+      if(metadataLocation == null) {
+        throw new IllegalArgumentException(format("missing metadata_location: %s.%s", database, tableName));
```

Comment:
I think this error message is unclear. It would be better to say that the table is missing its metadata location so it is clear what the object after `:` is.